fedn/network/clients/client_api.py

```diff
@@ -230,36 +336,34 @@ def send_status(self, msg: str, log_level=fedn.Status.INFO, type=None, request=N
         return self.grpc_handler.send_status(msg, log_level, type, request, sesssion_id, sender_name)
 
     def send_model_update(self,
-        sender_name: str,
-        sender_role: fedn.Role,
-        client_id: str,
         model_id: str,
         model_update_id: str,
-        receiver_name: str,
-        receiver_role: fedn.Role,
-        meta: dict
+        meta: dict,
+        request: fedn.TaskRequest
     ) -> bool:
+
         return self.grpc_handler.send_model_update(
-            sender_name=sender_name,
-            sender_role=sender_role,
-            client_id=client_id,
+            sender_name=self.name,
             model_id=model_id,
             model_update_id=model_update_id,
-            receiver_name=receiver_name,
-            receiver_role=receiver_role,
+            receiver_name=request.sender.name,
+            receiver_role=request.sender.role,
```

Comment:
Not sure about these changes... Don't know if a TaskRequest as input is the way to go. It creates a dependency that did not exist previously... If you think it's cleaner, keep it.